File: questions-dp600/fix_q4.py
with open('questions_4.js', 'r') as f:
    lines = f.readlines()

# Fix each line
fixed_lines = []
for line in lines:
    # Fix " explanation" -> "explanation"
    line = line.replace('", " explanation": "', '", "explanation": "')
    # Fix missing opening brace
    line = line.replace('\n  "num": 25, ', '\n  {"num": 25, ')
    # Fix options patterns like "B", HASH -> "B. HASH
    line = line.replace('", "B", ', '", "B. ')
    line = line.replace('", "C", ', '", "C. ')
    line = line.replace('", "D", ', '", "D. ')
    line = line.replace('", "b", ', '", "B. ')
    line = line.replace('", "c", ', '", "C. ')
    line = line.replace('", "d", ', '", "D. ')
    # Fix patterns like "A", at start of options
    line = line.replace('"options": ["a",', '"options": ["A.",')
    line = line.replace('"options": ["b",', '"options": ["B.",')
    line = line.replace('"options": ["c",', '"options": ["C.",')
    line = line.replace('"options": ["d",', '"options": ["D.",')
    # Fix patterns like "B", "HASH or "C", "VALIDATE
    line = line.replace('", "B", "', '", "B. "')
    line = line.replace('", "C", "', '", "C. "')
    line = line.replace('", "D", "', '", "D. "')
    fixed_lines.append(line)

with open('questions_4.js', 'w') as f:
    f.writelines(fixed_lines)

print('Fixed questions_4.js')

File: questions-dp600/fix_q4.js
const fs = require('fs');
let content = fs.readFileSync('questions_4.js', 'utf8');

// Fix " explanation" -> "explanation"
content = content.replace(/", " explanation": "/g, '", "explanation": "');

// Add missing { before "num": 25
content = content.replace(/\n  "num": 25, /g, '\n  {"num": 25, ');

// Fix options patterns - run multiple times to catch all variations
for (let i = 0; i < 3; i++) {
  content = content.replace(/"options": \[([^\]]+)\]/g, (match, opts) => {
    opts = opts.replace(/", "([A-D])", /g, '", "$1. ');
    opts = opts.replace(/", "([a-d])", /g, '", "$1. ');
    return '"options": [' + opts + ']';
  });
}

// Add missing closing }, after explanation before next {
content = content.replace(/("explanation": "[^"]+\.")\n  \{"num":/g, '$1,\n  {"num":');

fs.writeFileSync('questions_4.js', content);
console.log('Fixed questions_4.js');
